pkg/sql/opt/optbuilder/util.go

```diff
@@ -664,3 +664,30 @@ func findPublicTableColumnByName(tab cat.Table, name tree.Name) int {
 	}
 	return -1
 }
+
+type columnKinds struct {
+	// If true, include columns being added or dropped from the table. These
+	// are currently required by the execution engine as "fetch columns", when
+	// performing mutation DML statements (INSERT, UPDATE, UPSERT, DELETE).
+	includeMutations bool
+
+	// If true, include system columns.
+	includeSystem bool
+}
+
+// tableOrdinals returns a slice of ordinals that correspond to table columns of
+// the desired kinds.
+func tableOrdinals(tab cat.Table, k columnKinds) []int {
```

Comment:
Nice, I like the introduction of this. Should we pass in a slice in to write into or no?